src/print_graph.cc

```diff
@@ -154,18 +155,23 @@ namespace detail {
 		return label;
 	}
 
-	std::string print_command_graph(const command_graph& cdag, const task_manager& tm, const buffer_manager* const bm) {
+	std::string print_command_graph(const node_id local_nid, const command_graph& cdag, const task_manager& tm, const buffer_manager* const bm) {
 		std::string main_dot;
 		std::unordered_map<task_id, std::string> task_subgraph_dot;
 
+		const auto local_to_global_id = [local_nid](uint64_t id) {
+			// IDs in the DOT language may not start with a digit (unless the whole thing is a numeral)
+			return fmt::format("id_{}_{}", local_nid, id);
+		};
+
 		const auto print_vertex = [&](const abstract_command& cmd) {
 			static const char* const colors[] = {"black", "crimson", "dodgerblue4", "goldenrod", "maroon4", "springgreen2", "tan1", "chartreuse2"};
 
-			const auto name = cmd.get_cid();
-			const auto label = get_command_label(cmd, tm, bm);
-			const auto fontcolor = colors[cmd.get_nid() % (sizeof(colors) / sizeof(char*))];
-			const auto shape = isa<task_command>(&cmd) ? "box" : "ellipse";
-			return fmt::format("{}[label=<{}> fontcolor={} shape={}];", name, label, fontcolor, shape);
+			const auto id = local_to_global_id(cmd.get_cid());
+			const auto label = get_command_label(local_nid, cmd, tm, bm);
+			const auto* const fontcolor = colors[local_nid % (sizeof(colors) / sizeof(char*))];
+			const auto* const shape = isa<task_command>(&cmd) ? "box" : "ellipse";
```

Comment:
This `const auto` -> `const auto *const` substitution is a clang-tidy thing, isn't it? Because IMO it is actually worse for legibility and consistency.

Reply:
We'll remove this check, as per in-person discussion yesterday. I won't go back and undo all of these, though.

Reply:
Fair enough!